src/AzureExtension/DevBox/DevBoxInstance.cs

```diff
@@ -459,6 +463,14 @@ public ComputeSystemState GetState()
 
         _log.Information($"Getting State for devBox: '{DisplayName}', Provisioning: {provisioningState}, Action: {actionState}, Power: {powerState}");
 
+        // When a pinning operation has occured the Dev Box state will be set to
+        // Saving. In the future both a 'Pinning' and a 'Unpinning' member should
+        // be added to the 'ComputeSystemOperations' SDK enumeration object.

```

Comment:
I would argue that 'Pinning' and 'Unpinning' should not be part of the 'ComputeSystemOperations' SDK enumeration. Those operations are applied to the actual compute system, not to UX on the local client. Results of those operations are visible for all clients (DevHome and Web clients from different client devices) while 'Pinning' and 'Unpinning' only make sense for a particular client device. So logically those are different kind of operations.

If we need to add those operation type to the SDK I would add them as a separate enumeration (something like "ComputeSystemLocalOperations".

Reply:
fair I added this comment and the boolean `hasPinOperationStarted` originally. Huzaifa is adding it in his PR since I didn't have time to submit it separately. You make good points @huzaifa-d  lets go ahead and delete these comments.